### PR TITLE
Add getopt-for-visual-studio/20200201 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **getopt-for-visual-studio/20200201**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/getopt-for-visual-studio/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+class GetoptForVisualStudio(ConanFile):
+    name = "getopt-for-visual-studio"
+    description = "GNU getopt for Visual Studio"
+    topics = ("conan", "getopt", "cli", "command line", "options")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/skandhurkat/Getopt-for-Visual-Studi"
+    license = "MIT", "BSD-2-Clause"
+    exports_sources = "patches/**"
+    settings = "compiler"
```
Comment (uilianries) [inline recipes/getopt-for-visual-studio/all/conanfile.py]:
```suggestion
    settings = "compiler"
    no_copy_source = True
```

I think you can move the build step to package (patch)

Comment (madebr) [reply recipes/getopt-for-visual-studio/all/conanfile.py]: I added some patches. These won't work together with `no_copy_source=True`

**Comment by madebr** [comment]
@danimtb Please consider merging this one. :relaxed:

**Comment by danimtb** [comment]
Should we have this library when there is a clear deprecation notice at the author's repo?

https://github.com/skandhurkat/Getopt-for-Visual-Studio#deprication-notice

**Comment by madebr** [comment]
@danimtb 
Of course, the notics says to not use it in NEW libraries.
But older, mostly unix, libraries still use it.

**Pull request merged**